Added auto-reconnect & more commands

jcnlk opened the pull request.

jcnlk committed:
Added a simple settings menu

jcnlk committed:
Added Vigilance as required

jcnlk committed:
Added auto-reconnect & more commands

- Added auto-reconnect
- Added '/chat global'
- Added '/global connect/disconnect/help'
- Changed send command to '/globalchat [message]

jcnlk committed:
Update Backend to Host & remove unnecessary files

ActuallySilver commented on Frontend/Skyblock-Global-Chat-CT/metadata.json:
Add yourself as a creator.

ActuallySilver commented on Frontend/Skyblock-Global-Chat-CT/src/index.js:
```diff
@@ -1,7 +1,82 @@
-  event.setCancelled(true);
+
+  if (globalChatEnabled) {
+    const payload = {
```
Rename payload to messageData to make it more descript and also match backend

ActuallySilver commented on Frontend/Skyblock-Global-Chat-CT/src/index.js:
```diff
@@ -1,7 +1,82 @@
+});
+
+register("gameLoad", () => {
+  if (!World.isLoaded()) return;
```
Why does the world have to be loaded for us to connect?

ActuallySilver commented on Frontend/Skyblock-Global-Chat-CT/src/index.js:
```diff
@@ -1,7 +1,82 @@
+  }
+  const message = args.join(" ");
+
+  const payload = {
```
rename payload to messageData

ActuallySilver commented on Frontend/Skyblock-Global-Chat-CT/src/index.js:
```diff
@@ -1,7 +1,82 @@
+register("command", (...args) => {
+  if (!args) return;
+
+  const sub = (args[0] || "").toLowerCase();
```
does sub mean subCommand?
if so rename to subCommand
if not, i'm not sure what sub means? Variable names should be descriptive and not require outside information

ActuallySilver commented on Frontend/Skyblock-Global-Chat-CT/src/index.js:
```diff
@@ -1,7 +1,82 @@
+}).setName("global");
+
+register("command", (...args) => {
+  if (args.length > 0) {
```
return early and invert the boolean expression

ActuallySilver commented on Frontend/Skyblock-Global-Chat-CT/src/index.js:
```diff
@@ -1,7 +1,82 @@
+register("command", (...args) => {
+  if (args.length > 0) {
+    if (args[0].toLowerCase() !== "global") {
+      ChatLib.chat(`${ChatPrefix} ${RED}Usage: /chat global${RESET}`);
```
would this not get sent if someone was trying /chat sbe or /chat a?

ActuallySilver commented on Frontend/Skyblock-Global-Chat-CT/src/index.js:
```diff
@@ -1,7 +1,82 @@
+
+  ChatLib.chat(
+    `${ChatPrefix} ${GOLD}Auto-send to global is now:${RESET} ${
+      globalChatEnabled ? "§aEnabled§r" : "§cDisabled§r"
```
better to define things like this outside of the mesage.
add local variable for globalChatEnabledStatus and assign the value to there

ActuallySilver commented on Frontend/Skyblock-Global-Chat-CT/src/serverConnection.js:
```diff
@@ -1,22 +1,107 @@
+import { ChatPrefix, ConsolePrefix, RESET, YELLOW, RED, GREEN } from "./utils/constants";
 
-export const connection = new WebSocket("http://localhost:11000/");
+const WS_URL = "ws://localhost:11000/";
```
rename to GLOBAL_CHAT_WS_URL

ActuallySilver commented on Frontend/Skyblock-Global-Chat-CT/src/serverConnection.js:
```diff
@@ -1,22 +1,107 @@
+    this.autoReconnect = true;
+  }
+
+  send(jsonString) {
```
This should accept json and convert it into a string. the caller should not know what datatype is needed to be sent through the socket

ActuallySilver commented on Frontend/Skyblock-Global-Chat-CT/src/serverConnection.js:
```diff
@@ -1,22 +1,107 @@
+        }s...${RESET}`
+      );
+
+      this.reconnectTimer = setTimeout(() => {
```
does this work? I read that setTimeout doesn't work within chat triggers mods

ActuallySilver commented on Frontend/Skyblock-Global-Chat-CT/src/settings.js:
```diff
@@ -0,0 +1,46 @@
+        // this.addDependency("Setting1", "Setting2");
+
+
+        // Here category descriptions could be added
```
remove unneeded comments

ActuallySilver commented on Frontend/Skyblock-Global-Chat-CT/src/settings.js:
```diff
@@ -0,0 +1,46 @@
+    })
+    enableAutoconnect = true;
+
+    // more settings could be here
```
remove comment

ActuallySilver commented on Frontend/Skyblock-Global-Chat-CT/src/utils/constants.js:
```diff
@@ -0,0 +1,32 @@
+//const metadata = JSON.parse(read("Skyblock-Global-Chat", "metadata.json"));
+//
+//export const ModuleVersion = metadata.version;
```
remove comments if not needed

ActuallySilver reviewed (changes requested):
Added comments for you to address

jcnlk committed:
Renamed consts to be more descript

- Renamed consts to be more descript
- Changed '/chat global' to '/global toggle'
- Fixed crash which is caused by empty args
- Removed unnedded comments

jcnlk committed:
Added sln files to gitignore

jcnlk committed:
Update .gitignore

jcnlk committed:
Merge remote-tracking branch 'upstream/main'

jcnlk committed:
Updated .gitignore

ActuallySilver commented on Frontend/Skyblock-Global-Chat-CT/src/index.js:
```diff
@@ -1,7 +1,76 @@
+});
+
+register("gameLoad", () => {
+  //if (!World.isLoaded()) return;
```
Remove comment

ActuallySilver commented on Frontend/Skyblock-Global-Chat-CT/src/index.js:
```diff
@@ -1,7 +1,76 @@
+      break;
+    case "toggle":
+      globalChatEnabled = !globalChatEnabled;
+      const globalChatEnabledStatus = globalChatEnabled ? `${GREEN}Enabled${RESET}` : `${RED}Disabled${RESET}`;
```
change to 
```
globalChatEnabled = !globalChatEnabled;

const globalChatEnabledStatus = globalChatEnabled ? `${GREEN}Enabled${RESET}` : `${RED}Disabled${RESET}`;
ChatLib.chat(`${ChatPrefix} ${GOLD}Auto-send to global is now:${RESET} ${globalChatEnabledStatus}`);
break;
```

to make it more visible that globalChatEnabledStatus is used for the chat send (whitespace is useful if used correctly, and can confuse if used incorrectly)

ActuallySilver reviewed (changes requested):
few more tiny changes.

jcnlk committed:
Update index.js

jcnlk committed:
Better formatting
